ClassicAssist/Data/Filters/SoundFilter.cs

```diff
@@ -32,28 +32,62 @@
 
 namespace ClassicAssist.Data.Filters
 {
-    [FilterOptions( Name = "Sound Filter", DefaultEnabled = false )]
+    [FilterOptions(Name = "Sound Filter", DefaultEnabled = false)]
     public class SoundFilter : DynamicFilterEntry, IConfigurableFilter
     {
-        private const string DATA_FILE = "SoundFilters.json";
+        //private const string DATA_FILE = "SoundFilters.json";
 
         public SoundFilter()
         {
-            string dataPath = Path.Combine( Engine.StartupPath ?? Environment.CurrentDirectory, "Data" );
+            string dataPath = Path.Combine(Engine.StartupPath ?? Environment.CurrentDirectory, "Data\\Filters\\Audio\\");
+            //System.Windows.Forms.MessageBox.Show("" + dataPath);
+            ProcessDirectory(dataPath);
+        }
+
+        // Process all files in the directory passed in, recurse on any directories
+        // that are found, and process the files they contain.
+        public void ProcessDirectory(string targetDirectory)
+        { 
+            try
+            {
+
+                // Process the list of files found in the directory.
+                string[] fileEntries = Directory.GetFiles(targetDirectory);
+                foreach (string fileName in fileEntries)
+                {
+                    ProcessFile(fileName);
+                }
 
-            if ( !File.Exists( Path.Combine( dataPath, DATA_FILE ) ) )
+                // Recurse into subdirectories of this directory.
+                string[] subdirectoryEntries = Directory.GetDirectories(targetDirectory);
+                foreach (string subdirectory in subdirectoryEntries)
+                {
+                    ProcessDirectory(subdirectory);
+                }
+
+            }
+            catch (Exception ex)
+            {
+                System.Windows.Forms.MessageBox.Show("" + ex);
+            }
+        }
+
+        // Insert logic for processing found files here.
+        public void ProcessFile(string path)
+        {
+            if (!File.Exists(path))
             {
                 return;
             }
 
             Items.AddRange(
                 JsonConvert.DeserializeObject<SoundFilterEntry[]>(
-                    File.ReadAllText( Path.Combine( dataPath, DATA_FILE ) ) ) );
+                    File.ReadAllText(path)));
 
-            foreach ( SoundFilterEntry item in Items )
+            foreach (SoundFilterEntry item in Items)
             {
-                item.LocalizedName = Strings.ResourceManager.GetString( item.Name ) ?? item.Name;
-                item.Category = Strings.ResourceManager.GetString( item.Category ) ?? item.Category;
```

Comment:
Include spaces between parenthesis, don't modify lines you haven't changed

Reply:
Its VS that does it

Reply:
You need to set spaces between parenthesis in options

Reply:
Will do 👍